frontend/src/components/GalleryPageComponents/ImageDetailsOverlay.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import CloseRoundedIcon from '@mui/icons-material/CloseRounded';
+import WhatsAppIcon from '@mui/icons-material/WhatsApp';
+import FacebookIcon from '@mui/icons-material/Facebook';
+import TwitterIcon from '@mui/icons-material/Twitter';
+import "../../css/GalleryPageCSS/ImageDetailsOverlay.css";
+
+function ImageDetailsOverlay({ images }) {
+    const { imageId } = useParams(); // Extracting the `imageId` from the route parameters
+    const navigate = useNavigate(); // Navigation hook for programmatic navigation
+    const [copied, setCopied] = useState(false); // State to manage the "Copied" status for the URL
+    const image = images.find((img) => img.id === parseInt(imageId)); // Finding the specific image based on the `imageId`
+
+    if (!image) return <div>Image not found</div>; // If no matching image is found, display a fallback message
+
+    const copyToClipboard = () => {
+        navigator.clipboard.writeText(image.imgSrc);
+        setCopied(true);
+        setTimeout(() => setCopied(false), 2000);
```

Comment:
What is the purpose of this?

Reply:
The purpose of this was to copy the img url to clipboard.

Reply:
Yeah but why the timeout